### PR TITLE
Chrome and Firefox stack trace fix

### PR DESCRIPTION
**Pull request by KrishMunot**

### CODE REVIEW

File: packages/ember-debug/lib/deprecate.js [hunk 1]
```diff
@@ -105,6 +106,9 @@ if (DEBUG) {
           stack = error.stack.replace(/(?:\n@:0)?\s+$/m, '').
             replace(/^\(/gm, '{anonymous}(').split('\n');
         }
+      } finally {
+          Error.prepareStackTrace = prevPrepareStackTrace;
```
Comment (rwjblue) [inline packages/ember-debug/lib/deprecate.js]: Where is `prevPrepareStackTrace` being captured?

File: packages/ember-debug/lib/deprecate.js [hunk 2]
```diff
@@ -91,9 +91,10 @@ if (DEBUG) {
     if (ENV.LOG_STACKTRACE_ON_DEPRECATION) {
       let stackStr = '';
       let error = captureErrorForStack();
-      let stack;
-
-      if (error.stack) {
+     
+      // do this when we are actually processing the stack
+      try {
+      Error.prepareStackTrace = function(error, stack) {
```
Comment (rwjblue) [inline packages/ember-debug/lib/deprecate.js]: Indentation seems off here...

**Review by rwjblue — COMMENTED** [review]
Seems like this is roughly the same area as @stefanpenner's https://github.com/emberjs/ember.js/pull/15524, are y'all working on this together? Should we land them in any particular order?

**Comment by locks** [comment]
@KrishMunot if you could join efforts with @stefanpenner in https://github.com/emberjs/ember.js/pull/15524 it would be very helpful 🙏

**Pull request closed without merging**